ui/pages/confirmations/selectors/confirm.ts

```diff
@@ -21,22 +20,16 @@ export function pendingConfirmationsSelector(state: ConfirmMetamaskState) {
 export function pendingConfirmationsSortedSelector(
   state: ConfirmMetamaskState,
 ) {
-  return getPendingApprovals(state)
+  return Object.values(state.metamask.pendingApprovals ?? {})
     .filter(({ type }) =>
       ConfirmationApprovalTypes.includes(type as ApprovalType),
     )
     .sort((a1, a2) => a1.time - a2.time);
 }
 
-const internalLatestPendingConfirmationSelector = createSelector(
+export const oldestPendingConfirmationSelector = createDeepEqualSelector(
   pendingConfirmationsSortedSelector,
-  (pendingConfirmations) =>
-    pendingConfirmations.sort((a1, a2) => a2.time - a1.time)[0],
-);
-
-export const latestPendingConfirmationSelector = createDeepEqualSelector(
-  internalLatestPendingConfirmationSelector,
-  (latestPendingConfirmation) => latestPendingConfirmation,
+  (pendingConfirmations) => pendingConfirmations[0],
```

Comment:
Since the deep equal selector does the comparison on the dependencies, then this will still fire each time any of the confirmations change.

Ideally, we'd wrap the entire selector in a `createDeepEqualSelector`, or use `isEqual` as the second argument when we call `useSelector`.